lib/SVN/Notify.pm

```diff
@@ -1475,7 +1475,7 @@ sub output_headers {
         $norm = sub {
             return join ', ' => map {
                 my ($addr) = Email::Address::XS->parse($_);
-                if ($addr) {
+                if ($addr->is_valid()) {
```

Comment:
In scalar context since 1.01 it always returns Email::Address::XS object.
```
In scalar context this function returns just first parsed object. If more then one object was parsed then is_valid method on returned object returns false. If no object was parsed then empty Email::Address::XS object is returned.
```

@theory you are looking at old version of patches in this pull request
@mpghf Ideally squash changes, so there would not be any reverts or old versions

Reply:
Oh, missed that @pali, thanks.

Reply:
Running if($addr) on an empty address caused the warnings - I'm not familiar enough with perl's behaviour in that case to understand why.  

Using if(defined $addr) instead worked, but @pali has given a better option where $addr will never be undef.

Reply:
> Running if($addr) on an empty address caused the warnings

That is probably because of stringification operator which calls format() method? (Now just guessing).

Reply:
@pali Thanks.  Good to know.